app/src/main/java/by/lomazki/pokemontask5/model/Repository.kt

```diff
@@ -0,0 +1,55 @@
+package by.lomazki.pokemontask5.model
+
+import by.lomazki.pokemontask5.model.datasource.api.ApiService
+import by.lomazki.pokemontask5.model.datasource.room.PokemonDatabase
+import by.lomazki.pokemontask5.viewmodel.model.Pokemon
+
+
+class Repository(
+    private val database: PokemonDatabase,
+    private val apiService: ApiService
+) {
+
+    suspend fun getListPokemonApi(limit: Int, offset: Int = 0): List<Pokemon> {
+        var result = emptyList<Pokemon>()
+        runCatching {
+            apiService
+                .getPokemonApi()
+                .getGeneralRequest(limit, offset)
+                .pokemonList
+        }.onSuccess {
+            result = it
+        }.onFailure {
+            result = emptyList()    // TODO вывести Error
+        }
+        return result
```

Comment:
```suggestion
        return runCatching {
            apiService
                .getPokemonApi()
                .getGeneralRequest(limit, offset)
                .pokemonList
        }.fold(onSuccess = {
            it
        },onFailure = {
            emptyList()
        })
```
или даже так 
`runCatching { request() }.successOrNull() ?: emptyList()`